### PR TITLE
TIP-121c: vlAURA OTC proposal

### PR DESCRIPTION
**Pull request by eswak**
Add a ProxyOTCEscrow contract that will receive proxy ownership from Tribe DAO in order to allow an external entity (OTC buyer) to purchase the proxy admin.

- [x] All Tests Passing
- [x] Proposal Added to ProposalsConfig
- [x] Fork Block Correct
- [x] Remove Any .only's on Tests
- [x] Deploy Contract If Needed (+ Verify on Etherscan & Add to MainnetAddresses)
- [x] Update Documentation If Needed
- [x] Update Roles Config
- [x] Update Collateralization Oracle If Needed
- [ ] Proposal Submitted

### CODE REVIEW

File: proposals/dao/tip_auraotc.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+const fipNumber = 'auraotc';
+
+let auraOtcBuyer: string;
+let pcvStatsBefore: PcvStats;
+
+// Do any deployments
+// This should exclusively include new contract deployments
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  auraOtcBuyer = addresses.eswak;
```
Comment (eswak) [inline proposals/dao/tip_auraotc.ts]: Todo: update with Fishy's address

File: proposals/dao/tip_auraotc.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  return {
+    vlauraOtcHelper
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  auraOtcBuyer = addresses.eswak;
```
Comment (eswak) [inline proposals/dao/tip_auraotc.ts]: Todo: update with Fishy's address

**Comment by eswak** [comment]
Rebased on develop to get the latest changes & remove veBAL OTC commits

**Review by thomas-waite — COMMENTED** [review]
How will Fishy manage the AURA - the deposit implementation has various management methods permissioned to TribeRoles. Will he need to upgrade the implementation?

Would prefer we test the `recoverProxyOwnership()` but on inspection it looks fine.

Deployed contract looks good

**Review by thomas-waite — DISMISSED** [review]
Questions answered

**Comment by eswak** [comment]
Calldata : 
```
0x7d5e81e2000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001c00000000000000000000000000000000000000000000000000000000000000001000000000000000000000000f8c2b645988b7658e7748ba637fe25bdd46a704a000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000447eff275e000000000000000000000000c44902c03093d52213d20e5b06a0bda4d9ce652400000000000000000000000061a991153e3ff68844abce91551e046e1be6f7640000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000b15449502d313231633a20766c41555241204f54430a0a20205449502d313231633a20766c41555241204f54430a0a2020546869732070726f706f73616c2073657473207570207468652070726f746f636f6c20746f2062652061626c6520746f207472616e73666572206974732033342c30333820766c4155524120746f20466973687920666f72204f54432074726164652c20696e2065786368616e6765206f662039342c303030204441492e0a2020000000000000000000000000000000
```

**Comment by eswak** [comment]
I refreshed the branch to solve conflicts with develop; calldata hasn't changed

Fishy gave the greenlight to ship

File: contracts/metagov/VlAuraFishyImplementation.sol [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        address token
+    ) external view returns (address bribeToken, uint256 bribeAmount);
+}
+
+/// @author eswak
+contract VlAuraFishyImplementation {
+    address public constant AURA = 0xC0c293ce456fF0ED870ADd98a0828Dd4d2903DBF;
+    address public constant VLAURA = 0x3Fa73f1E5d8A792C80F426fc8F84FBF7Ce9bBCAC;
+
+    // TODO: Update this before deploying, Fishy.
```
Comment (thomas-waite) [inline contracts/metagov/VlAuraFishyImplementation.sol:24]: Flagging this TODO

File: contracts/metagov/VlAuraFishyImplementation.sol [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    address public constant OWNER = 0x6ef71cA9cD708883E129559F5edBFb9d9D5C6148;
+
+    modifier onlyOwner() {
```
Comment (thomas-waite) [inline contracts/metagov/VlAuraFishyImplementation.sol:27]: Will he ever need to change the owner? What if he sells this on and just wants to give the new owner control over this implementation contract without upgrading? Why not just import OZ's ownable library so you have `transferOwner()`?

File: contracts/metagov/VlAuraFishyImplementation.sol [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyOwner {
+        IERC20(token).transfer(to, amount);
+    }
+
+    /// @notice forward rewards to an eoa on Hidden Hand
+    function setRewardForwarding(address briber, address to) external onlyOwner {
+        // note: briber for aura is currently 0x642c59937A62cf7dc92F70Fd78A13cEe0aa2Bd9c
```
Comment (thomas-waite) [inline contracts/metagov/VlAuraFishyImplementation.sol:70]: Maybe put this comment in a `@dev` natspec just above the function?

File: contracts/metagov/VlAuraFishyImplementation.sol [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        IAuraBribe(briber).setRewardForwarding(to);
+    }
+
+    /// @notice arbitrary call
+    function call(
```
Comment (thomas-waite) [inline contracts/metagov/VlAuraFishyImplementation.sol:75]: Great idea - will be able to call to any method not currently exposed

**Review by thomas-waite — COMMENTED** [review]
Should the implementation contract have more flexibility about who the owner is? Why not import OZ Ownable and then there will be a `transferOwner()` method if it's needed for whatever reason?

File: contracts/metagov/VlAuraFishyImplementation.sol (hunk 3 above)
Comment (eswak) [reply contracts/metagov/VlAuraFishyImplementation.sol:24]: it was for him to try the tests on his own ^^

File: contracts/metagov/VlAuraFishyImplementation.sol (hunk 4 above)
Comment (thomas-waite) [reply contracts/metagov/VlAuraFishyImplementation.sol:27]: As discussed, he's happy with the current version

**Review by thomas-waite — APPROVED** [review]
Looks good

**Pull request merged**